stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/io/ReadCharacters.java

```diff
@@ -55,29 +65,81 @@
      */
     private static final int NUMBER_OF_CHARS_INDEX = 0;
 
+    /**
+     * Will be the I/O event handler.
+     */
+    private EventManager eventManager = EventManager.getInstance();
+
+    private static final Logger log = LoggerFactory.getLogger(ReadCharacters.class);
+
+    /*
+     * Callback method of the read characters response.
+     *
+     * @param result the result returned as the response.
+     * @return the processed event result.
+     *//*
+    private static EventResult readCharactersResponse(EventResult<Boolean, EventContext> result) {
+        BStruct errorStruct;
+        EventContext eventContext = result.getContext();
+        Context context = eventContext.getContext();
+        Throwable error = eventContext.getError();
+        if (null != error) {
+            errorStruct = IOUtils.createError(context, error.getMessage());
+        }
+        Boolean numberOfBytes = result.getResponse();
+        return result;
+    }
+    */
+
+    /**
+     * Reads characters asynchronously.
+     *
+     * @param numberOfCharacters number of characters which should be read.
+     * @param characterChannel   the channel which the characters will be read.
+     * @param context            context of the event.
+     * @return the content which is read.
+     * @throws ExecutionException   if an error occurs in the async framework.
+     * @throws InterruptedException during interrupt.
+     */
+    private String readCharacter(int numberOfCharacters, CharacterChannel characterChannel, EventContext context) throws
+            ExecutionException, InterruptedException {
+        ReadCharactersEvent event = new ReadCharactersEvent(characterChannel, numberOfCharacters, context);
+        CompletableFuture<EventResult> future = eventManager.publish(event);
+        EventResult eventResult = future.get();
+        Throwable error = ((EventContext) eventResult.getContext()).getError();
+        if (null != error) {
+            throw new ExecutionException(error);
+        }
+        return (String) eventResult.getResponse();
+    }
+
     /**
      * <p>
      * Reads characters from the channel.
      * </p>
-     *
+     * <p>
      * {@inheritDoc}
      */
     @Override
     public BValue[] execute(Context context) {
         BStruct channel;
         long numberOfCharacters;
-        BString content;
+        BString content = null;
```

Comment:
Shall we define above 2 variables (channel, numberOfCharacters) within the try block it self. That will reduce them to a single line and more readable

Reply:
In terms of readability, having variables defined outside the try block would help understand it's usage, and the types of memory allocated for the whole process. And some of the variables are being used by catch blocks etc which will not allow to move them under the try scope